tests/unit/resources/test_factory.py

```diff
@@ -358,12 +360,14 @@ def test_resource_lazy_loads_properties(self, action_cls):
                 }
             }
         }
-        shape = mock.Mock()
-        shape.members = {
-            'Url': None,
-            'ETag': None,
-            'LastModified': None,
-        }
+        shape = StructureShape('TestShape', {
+            'type': 'structure',
+            'members': {}
+        })
+        shape.members['ETag'] = Shape('ETag', {'type': 'string'})
```

Comment:
This is relying on an implementation detail that is subject to change.  `members` is actually a property (a cached property) that happens to create a mutable attr the first time it's retrieved.

Alternatives:

* Pass in a `ShapeResolver` with a `shape_map`.
* Use the botocore.model.DenormalizedStructureBuilder.